book/core-concepts/aggregations.md

```diff
@@ -0,0 +1,68 @@
+# Aggregations
+
+Aggregations are an alternative to state computations that trade some of the freedom provided by state computations for the ability to efficiently compute results in windows. 
+
+An aggregation consists of four parts:
+1) The initial accumulator state.
+2) An `update` function that takes an input and the accumulator state and optionally updates the accumulator.
+3) A `combine` function that takes two accumulators and returns an accumulator that represents the combine of both of them.
+4) An `output` function that takes a key and an accumulator and optionally outputs an aggregation result.
+
+Let's look at the example of a user-defined `Sum` aggregation, using the Python API (the API-specific details are explained [here](/book/python/api.md); we're taking a higher level view at the moment). We first define a helper class `_Sum` which is used as our accumulator:
+
+```python
+class _Sum():
+    def __init__(self, initial_total):
+        self.total = initial_total
+
+    def add(self, value):
+        self.total = self.total + value
+
+class Sum(wallaroo.Aggregation):
+    def initial_accumulator(self):
+        return _Sum(0)
+
+    def update(self, input, sum):
+        sum.add(sum.total + input.value)
```

Comment:
should this be `sum.add(input.value)`?
`sum.total + value` is already done inside of `_Sum.add`

Reply:
addressed